ml_peg/analysis/utils/utils.py

```diff
@@ -3,17 +3,86 @@
 from __future__ import annotations
 
 from collections.abc import Callable
+from pathlib import Path
 from typing import Any
 
 from matplotlib import cm
 from matplotlib.colors import Colormap
 import numpy as np
 from scipy.stats import rankdata
 from sklearn.metrics import mean_absolute_error, mean_squared_error
+from yaml import safe_load
 
 from ml_peg.app.utils.utils import clean_weights
 
 
+def load_metrics_config(config_path: Path) -> tuple[dict[str, Any], dict[str, str]]:
+    """
+    Load metric thresholds and tooltips from a YAML configuration file.
+
+    Parameters
+    ----------
+    config_path
+        Path to the YAML file containing metric definitions.
+
+    Returns
+    -------
+    tuple[dict[str, Any], dict[str, str]]
+        Mapping of metric thresholds and mapping of metric tooltips.
+    """
+    if not config_path.exists():
+        msg = f"Metrics configuration file not found: {config_path}"
+        raise FileNotFoundError(msg)
+
+    with config_path.open(encoding="utf8") as stream:
+        data = safe_load(stream) or {}
+
+    metrics_data = data.get("metrics", {}) or {}
+    if not isinstance(metrics_data, dict):
+        msg = "Expected 'metrics' section in metrics config to be a mapping."
+        raise TypeError(msg)
+    thresholds: dict[str, Any] = {}
+    tooltips: dict[str, str] = {}
+
+    for metric_name, metric_config in metrics_data.items():
+        if not isinstance(metric_config, dict):
+            msg = f"Metric configuration for '{metric_name}' must be a mapping."
+            raise TypeError(msg)
+
+        unit = metric_config.get("unit", metric_config.get("units"))
+        level = metric_config.get("level_of_theory", metric_config.get("level"))
```

Comment:
Same as above. Maybe just `theory`? I don't think `level` is particularly clear on its own